contrib/buildrefactor/tests/python/pants_test/contrib/buildrefactor/test_buildozer.py

```diff
@@ -43,43 +46,49 @@ def test_remove_single_dependency(self):
   def test_remove_multiple_dependencies(self):
     self._test_remove_dependencies('d', 'a b')
 
+  def test_custom_command_error(self):
+    with self.assertRaises(TaskError):
+      self._test_buildozer_execution({ 'command': 'foo', 'add-dependencies': 'boo' })
+
   def test_custom_command(self):
-    build_file = self.build_root + '/b/BUILD'
+    build_file = '{}/b/BUILD'.format(self.build_root)
     new_build_name = 'b_2'
 
-    self._clean_build_file(build_file)
+    clean_build_file(build_file)
     self._test_buildozer_execution({ 'command': 'set name {}'.format(new_build_name) })
+    assertInFile(self, new_build_name, build_file)
 
-    with open(build_file) as f:
-      build_source = f.read()
+  def test_execute_binary(self):
+    init_subsystem(BinaryUtil.Factory)
 
-    self.assertIn(new_build_name, build_source)
+    build_file = '{}/b/BUILD'.format(self.build_root)
+    new_build_name = 'b_2'
 
-  def test_custom_command_error(self):
-    with self.assertRaises(TaskError):
-      self._test_buildozer_execution({ 'command': 'foo', 'add-dependencies': 'boo' })
+    clean_build_file(build_file)
+    Buildozer.execute_binary('set name {}'.format(new_build_name), address = Address.parse('b'))
+    assertInFile(self, new_build_name, build_file)
 
   def _test_add_dependencies(self, spec_path, dependencies_to_add):
-    build_file = self.build_root + '/{}/BUILD'.format(spec_path)
+    build_file = '{}/{}/BUILD'.format(self.build_root, spec_path)
 
-    self._clean_build_file(build_file)
+    clean_build_file(build_file)
     self._test_buildozer_execution({ 'add_dependencies': dependencies_to_add })
 
     for dependency in dependencies_to_add.split(' '):
       self.assertIn(dependency, self._build_file_dependencies(build_file))
 
   def _test_remove_dependencies(self, spec_path, dependencies_to_remove):
-    build_file = self.build_root + '/b/BUILD'
+    build_file = '{}/{}/BUILD'.format(self.build_root, spec_path)
 
-    self._clean_build_file(build_file)
-    self._test_buildozer_execution({ 'remove_dependencies': dependencies_to_remove })
+    clean_build_file(build_file)
+    self._test_buildozer_execution({ 'remove_dependencies': dependencies_to_remove }, spec_path)
 
     for dependency in dependencies_to_remove.split(' '):
       self.assertNotIn(dependency, self._build_file_dependencies(build_file))
 
-  def _test_buildozer_execution(self, options):
+  def _test_buildozer_execution(self, options, root_target='b'):
```

Comment:
this method does not do any testing/assertion but only execute the task. can probably rename to `run_buildozer_task` for example.